### PR TITLE
Feature/jel 306 rspec provisioning tests

### PR DESCRIPTION
**Pull request by jntullo**
Let's discuss @brittballard

### CODE REVIEW

**Comment by chriskacerguis** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/54d4fa64695680762a0024c5/compare/feature/jel_306_rspec_provisioning_tests">analyzed this pull request</a>.

File: spec/workers/aws_fog_spec.rb [hunk 1]
```diff
@@ -1,26 +1,84 @@
...
+      { product_type_id: ProductType.find_by(name: 'Databases').id, manageiq_key: 'db_instance_class' },
+      { product_type_id: ProductType.find_by(name: 'Databases').id, manageiq_key: 'engine' },
+      { product_type_id: ProductType.find_by(name: 'Databases').id, manageiq_key: 'storage_type' }
+    ])
+    ProductAnswer.create([
+      { product_id: Product.find_by(name: 'rspec_product').id, product_type_question_id: ProductTypeQuestion.find_by(manageiq_key: 'allocated_storage').id, answer: '5' },
+      { product_id: Product.find_by(name: 'rspec_product').id, product_type_question_id: ProductTypeQuestion.find_by(manageiq_key: 'db_instance_class').id, answer: 'db.m3.medium' },
+      { product_id: Product.find_by(name: 'rspec_product').id, product_type_question_id: ProductTypeQuestion.find_by(manageiq_key: 'engine').id, answer: 'mysql' },
+      { product_id: Product.find_by(name: 'rspec_product').id, product_type_question_id: ProductTypeQuestion.find_by(manageiq_key: 'storage_type').id, answer: 'standard' }
+    ])
```
Comment (brittballard) [inline spec/workers/aws_fog_spec.rb]: I would recommend using factories to do this. I use them in some of the examples specs. We can discuss in a remote pair in more detail.

File: spec/workers/aws_fog_spec.rb [hunk 2]
```diff
@@ -1,26 +1,84 @@
...
+    AwsFog.new(order_item.id).retire
+    order_item.reload
+    expect(order_item.provision_status). to eq 'retired'
+  end
 
+  def prepare_fog_spec(name)
+    enable_aws_fog_provisioning
     create(
       :order_item,
       product: create(
```
Comment (brittballard) [inline spec/workers/aws_fog_spec.rb]: These `create` calls are using factories.

File: spec/workers/aws_fog_spec.rb [hunk 3]
```diff
@@ -1,26 +1,84 @@
...
+      prepare_db_products
+      provision_product order_item
+      retire_product order_item
+    end
+
+    it 'retires storage using fog' do
+      order_item = prepare_fog_spec 'Storage'
+      provision_product order_item
+      retire_product order_item
+    end
```
Comment (brittballard) [inline spec/workers/aws_fog_spec.rb]: In these we should see if we can come up with a good `expect` check to use.

**Comment by brittballard** [comment]
@jntullo these look good. What time would work for you to discuss via remote pair today?

**Comment by chriskacerguis** [comment]
@brittballard @jntullo what's the status here, can we merge?

**Comment by jntullo** [comment]
@chriskacerguis will be doing an updated commit later. None of this changes. Up to you, @brittballard

**Comment by chriskacerguis** [comment]
@jntullo hmm..
@OBCENEIKON is looking into this

**Comment by brittballard** [comment]
@jntullo what do you think of using the factories for the settings stuff?

**Comment by jntullo** [comment]
@brittballard yeah I can do that!

File: spec/workers/aws_fog_spec.rb (hunk 1 above)
Comment (jntullo) [inline spec/workers/aws_fog_spec.rb]: @brittballard having difficulty creating factories for this

**Comment by jntullo** [comment]
@brittballard let me know what you think.

**Comment by brittballard** [comment]
That's some final boss level testing there. Haha, looks great to me.

**Comment by jntullo** [comment]
closes #306

**Comment by jntullo** [comment]
All of this is now irrelevant with the new AWS fog module.

**Comment by brittballard** [comment]
@jntullo we should still be able to use the tests. We're still trying to workout the best way though.

**Comment by jntullo** [comment]
@brittballard ok cool! Let me know how I can help

**Pull request closed without merging**